src/documentation/orogen/plugins.page

```diff
@@ -29,12 +29,28 @@ seamless integration as it is the only one that has an API for code generation.
 Example of a very simple plugin:
 
 ~~~ ruby
-module OrogenSimplePlugin
-    def add_boolean_attribute(name)
-        add_base_member("simple_plugin", name, "bool")
+class BoolPlugin <  OroGen::Spec::TaskModelExtension
+    attr_accessor :varname
+
+    def early_register_for_generation(task)
+        task.add_base_member("simple_plugin", varname, "bool")
+        task.operations["getBool"].
+            base_body("return #{varname};")
+    end
+end
+
+class Orocos::Spec::TaskContext
+    def add_boolean_attribute(varname)
+        extension_name = "BoolPlugin"
+        if !find_extension(extension_name)
```

Comment:
If you want that your plugin to only change the task once, this is not a bad idea. It prevents multiple instances. I can change it that way, that you get the message "there is already an extension called" when trying to instantiate the same plugin twice.

Basically the question is: Do we want to show a plugin that can create multiple members and methods or a plugin that effects the task only once?